sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -79,7 +79,9 @@ abstract class QueryStageExec extends LeafExecNode {
    */
   def computeStats(): Option[Statistics] = resultOption.map { _ =>
     // Metrics `dataSize` are available in both `ShuffleExchangeExec` and `BroadcastExchangeExec`.
-    Statistics(sizeInBytes = plan.metrics("dataSize").value)
+    // The `dataSize` metric may be a negative number when this spark plan generates empty result,
+    // since SQLMetric use -1 as initial value.
+    Statistics(sizeInBytes = Math.max(plan.metrics("dataSize").value, 0))
```

Comment:
This is a bug of the `dataSize` metrics, not AQE, let's fix the underlying issue.

Can you take over https://github.com/apache/spark/pull/25095 ?

Reply:
OK, I'll take over it